src/main/java/org/folio/rest/impl/TenantPermsAPI.java

```diff
@@ -94,21 +94,28 @@ public void postTenantpermissions(OkapiPermissionSet entity, Map<String, String>
       Context vertxContext, String tenantId) {
     return getPermsByModule(moduleId, vertxContext, tenantId)
         .compose(existing -> {
-          if (existing.isEmpty() && perms != null) {
-            // this means either:
-            // A) the first time enabling this module, or
-            // B) the permissions exist but don't yet have the moduleName field
-            List<Permission> ret = new ArrayList<>();
-            List<Future> futures = new ArrayList<>(perms.size());
-            perms.forEach(perm ->
+          if (!existing.isEmpty()) {
+            return Future.succeededFuture(existing); // permission already has context
+          }
+          // this means either:
+          // A) the first time enabling this module, or
+          // B) the permissions exist but don't yet have the moduleName field
+          List<Permission> ret = new ArrayList<>();
+          List<Future> futures = new ArrayList<>(perms.size());
+          perms.forEach(perm ->
```

Comment:
possible NPE if perms == null...  IIRC there were unit tests which exercised the ability for perms == null, though in practice I don't think it's likely to happen.

Reply:
It's not going to happen as `handlePermLists` passes a good list. All parameters to `getPermsForModule` are expected to be non-null, but don't check for those either (and shouldn't).